### PR TITLE
Minor nits fast follow up to PR 903

### PR DESCRIPTION
**Pull request by nithu0115**
*Issue #, if available:* N/A

*Description of changes:*
This is a fast follow up to #903. This has clean up and minor nits.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/awsutils/awsutils.go [hunk 1]
```diff
@@ -248,10 +248,7 @@ func (ss *StringSet) Set(items []string) {
 }
 
 func (ss *StringSet) IsEmpty() bool {
-	if ss.data.Len() == 0 {
-		return true
-	}
-	return false
+	return ss.data != nil && ss.data.Len() == 0
```
Comment (anguslees) [inline pkg/awsutils/awsutils.go:251]: I think this method needs to hold the lock too (there are potential races between the two references to `ss.data`, and (possibly) internally in `Len()`)

File: pkg/awsutils/awsutils.go [hunk 2]
```diff
@@ -248,10 +248,7 @@ func (ss *StringSet) Set(items []string) {
...
 }
 
 func (ss *StringSet) Difference (other *StringSet) *StringSet {
```
Comment (anguslees) [inline pkg/awsutils/awsutils.go:254]: I think this method needs to hold the `other` lock too.  In this simple case/type, I think that's fine, but be aware that could lead to fun deadlocks if we get more complicated interactions between two StringSets.

Comment (anguslees) [reply pkg/awsutils/awsutils.go:254]: Fwiw, I think several of the uses of Difference below could instead use a regular (non-threadsafe) `sets.String`, and avoid thinking about locking.

Comment (nithu0115) [reply pkg/awsutils/awsutils.go:254]: Make sense.  I thought about using `sets.String` but after several discussions, I went with `StringSet` approach which I thought was much cleaner.

File: pkg/awsutils/awsutils.go (hunk 1 above)
Comment (nithu0115) [reply pkg/awsutils/awsutils.go:253]: Updated to hold a Rlock here.

File: pkg/awsutils/awsutils.go (hunk 2 above)
Comment (nithu0115) [reply pkg/awsutils/awsutils.go:256]: > I think this method needs to hold the `other` lock too. In this simple case/type, I think that's fine, but be aware that could lead to fun deadlocks if we get more complicated interactions between two StringSets.

Added `other.RLock`

**Pull request merged**